patches/net/minecraft/world/item/ItemStack.java.patch

```diff
@@ -287,3 +287,16 @@
      public void onDestroyed(ItemEntity p_150925_) {
          this.getItem().onDestroyed(p_150925_);
      }
+@@ -1016,6 +_,12 @@
+ 
+     public SoundEvent getEatingSound() {
+         return this.getItem().getEatingSound();
++    }
++
++    @Override
++    public int getBurnTime(@org.jetbrains.annotations.Nullable net.minecraft.world.item.crafting.RecipeType<?> recipeType) {
```

Comment:
The body should be in `IItemStackExtension`.

Reply:
The `ItemStack` function can perform extra logic. The `Item` method is only for overrides, not for queries.